python/mxnet/rnn/io.py

```diff
@@ -58,6 +62,8 @@ def encode_sentences(sentences, vocab=None, invalid_label=-1, invalid_key='\n',
     if vocab is None:
         vocab = {invalid_key: invalid_label}
         new_vocab = True
+    elif unknown_token:
+        new_vocab = True
```

Comment:
why?

Reply:
Hi, there is situation where users has their own dictionary, say `dict = {'a':1, 'b':2, 'c':3}` 'abc' are frequent tokens the user care about. All the rest rare tokens are considered as unknown token (say the user define it as `'UNK'`), that return a encoded list `[[1,2,3],[2,3,0]]`, a key-value pair `'UNK': 0` is added into the dictionary. 

But the previous version will raise error for this case, which by default assuming that user will provdie a thorough vocaburary.

Reply:
you should change the assertion to ignore cases where unkown_token is give instead of changing new_vocab to true

Reply:
Ping

Reply:
Sorry for late reply (in exam). Just fix according to your suggestion.